processor/groupbyattrsprocessor/processor_test.go

```diff
@@ -0,0 +1,213 @@
+	return res
+}
+
+func filterAttributeMap(attrMap pdata.AttributeMap, selectedKeys []string) pdata.AttributeMap {
+	filteredAttrMap := pdata.NewAttributeMap()
+	for _, key := range selectedKeys {
+		val, _ := attrMap.Get(key)
+		filteredAttrMap.Insert(key, val)
+	}
+	filteredAttrMap.Sort()
+	return filteredAttrMap
+}
+
+func TestLogAttributesExtracted(t *testing.T) {
+	tests := []struct {
+		name           string
+		groupByKeys    []string
+		nonGroupedKeys []string
+		count          int
+	}{
+		{
+			name:           "Two groupByKeys",
+			groupByKeys:    []string{"xx", "yy"},
+			nonGroupedKeys: []string{},
+			count:          4,
+		},
+		{
+			name:           "One attribute",
+			groupByKeys:    []string{"xx"},
+			nonGroupedKeys: []string{"yy"},
+			count:          4,
+		},
+		{
+			name:           "No groupByKeys",
+			groupByKeys:    []string{"zz"},
+			nonGroupedKeys: []string{"xx", "yy"},
+			count:          4,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			logs := someLogs(attrMap, tt.count)
+			spans := someSpans(attrMap, tt.count)
+
+			gap, err := createGroupByAttrsProcessor(logger, tt.groupByKeys)
+			assert.NoError(t, err)
+
+			expectedResource := prepareResource(attrMap, tt.groupByKeys)
+			expectedAttributes := filterAttributeMap(attrMap, tt.nonGroupedKeys)
+
+			processedLogs, err := gap.ProcessLogs(context.Background(), logs)
+			assert.NoError(t, err)
+
+			processedSpans, err := gap.ProcessTraces(context.Background(), spans)
+			assert.NoError(t, err)
+
+			assert.Equal(t, 1, processedLogs.ResourceLogs().Len())
+			assert.Equal(t, 1, processedSpans.ResourceSpans().Len())
+
+			resources := []pdata.Resource{
+				processedLogs.ResourceLogs().At(0).Resource(),
+				processedSpans.ResourceSpans().At(0).Resource(),
+			}
+
+			for _, res := range resources {
+				res.Attributes().Sort()
+				assert.Equal(t, expectedResource, res)
+			}
+
+			ills := processedLogs.ResourceLogs().At(0).InstrumentationLibraryLogs()
+			ilss := processedSpans.ResourceSpans().At(0).InstrumentationLibrarySpans()
+
+			assert.Equal(t, 1, ills.Len())
+			assert.Equal(t, 1, ilss.Len())
+
+			ls := ills.At(0).Logs()
+			ss := ilss.At(0).Spans()
+			assert.Equal(t, tt.count, ls.Len())
+			assert.Equal(t, tt.count, ss.Len())
+
+			for i := 0; i < ls.Len(); i++ {
+				log := ls.At(i)
+				span := ss.At(i)
+
+				log.Attributes().Sort()
+				span.Attributes().Sort()
+
+				assert.Equal(t, expectedAttributes, log.Attributes())
+				assert.Equal(t, expectedAttributes, span.Attributes())
+			}
+		})
+	}
+
+}
+
+/*func TestProcessorCapabilities(t *testing.T) {
+	// prepare
+	config := Config{
+		WaitDuration: time.Nanosecond,
+		NumTraces:    10,
+	}
+	st := newMemoryStorage()
+	next := &mockProcessor{}
+
+	// test
+	p, err := newGroupByTraceProcessor(logger, st, next, config)
+	caps := p.GetCapabilities()
+
+	// verify
+	assert.NoError(t, err)
+	assert.NotNil(t, p)
+	assert.Equal(t, true, caps.MutatesConsumedData)
+}*/
```

Comment:
Why is that commented out?

Reply:
Copypasta monster, was going to include the same case :)